client/src/taskProvider/HardhatTaskProvider.ts

```diff
@@ -46,10 +47,19 @@ export class HardhatTaskProvider implements vscode.TaskProvider {
           continue;
         }
 
+        let relativeProjectDir = projectDir;
+
+        for (const workspaceFolder of vscode.workspace.workspaceFolders || []) {
+          relativeProjectDir = relativeProjectDir.replace(
+            path.dirname(workspaceFolder.uri.fsPath) + path.sep,
+            ""
+          );
+        }
+
         const task = new Task(
           { type: TASK_TYPE, task: taskDef.name },
           vscode.TaskScope.Workspace,
-          `${taskDef.name} - ${projectDir}`,
+          `${taskDef.name} - ${relativeProjectDir}`,
           SOURCE,
           new vscode.ShellExecution("npx", ["hardhat", taskDef.command], {
```

Comment:
Does the dependency on npx imply we should turn flattening off if the file not in a hardhat project?